src/coreclr/src/zap/zapinfo.cpp

```diff
@@ -2182,7 +2192,20 @@ DWORD FilterNamedIntrinsicMethodAttribs(ZapInfo* pZapInfo, DWORD attribs, CORINF
                     fTreatAsRegularMethodCall = !fIsGetIsSupportedMethod;
                 }
             }
+#endif // defined(TARGET_X86) || defined(TARGET_AMD64)
+#ifdef TARGET_X86
+            else if (strcmp(className, "X64") == 0)
+            {
```

Comment:
If this isn't defensive enough, we might want to consider explicitly adding LZCNT and SSE3-4 checks so their x64 calls get trimmed in x86.